src/main.ts

```diff
@@ -409,7 +409,10 @@ function makeTimestampMethods(options: Options, longs: ReturnType<typeof makeLon
     seconds = code`${longs.numberToLong}(date.getTime() / 1_000)`;
   } else if (options.forceLong === LongOption.STRING) {
     toNumberCode = 'Number(t.seconds)';
-    seconds = '(date.getTime() / 1_000).toString()';
+    // Must discard the fractional piece here
+    // Otherwise the fraction ends up on the seconds when parsed as a Long
+    // (note this but would only occurs when the string is > 8 characters)
```

Comment:
Great comment, thanks!

Reply:
Aww man, you commented on it and now I noticed the horrible grammar.

Reply:
Resolved grammar issues in https://github.com/stephenh/ts-proto/pull/281/files

Reply:
Ha, I hadn't actually noticed :-) but thanks!